openedx/core/djangoapps/geoinfo/middleware.py

```diff
@@ -31,7 +29,7 @@ def process_request(self, request):
 
         Store country code in session.
         """
-        new_ip_address = get_real_ip(request)
+        new_ip_address, _ = get_client_ip(request, request_header_order=['X_FORWARDED_FOR', 'HTTP_X_FORWARDED_FOR'])
```

Comment:
Old method new_ip_address was appearing as `None` for few test cases. but in new method it returns `127.0.0.1` and those tests fail.